src/Control/Session.php

```diff
@@ -307,7 +312,16 @@ public function start(HTTPRequest $request)
         $data = [];
         if (!session_id() && (!headers_sent() || $this->requestContainsSessionId($request))) {
             if (!headers_sent()) {
-                session_set_cookie_params($timeout ?: 0, $path, $domain ?: null, $secure, true);
+                $sameSite = static::config()->get('cookie_samesite');
+                $secure = $this->isCookieSecure($sameSite, Director::is_https($request));
+                session_set_cookie_params([
+                    'lifetime' => $timeout ?: 0,
```

Comment:
We can add unit testing for this (and CookieJar::outputCookie) by creating private functions to generate the arrays to pass to session_set_cookie_params/setcookie and then using reflection to set their visibility to public

Reply:
Done. For `CookieJar::outputCookie()` the only piece of actual functionality there was whether secure is true or false, so I've extracted only that part into its own private method.